### ISSUE TITLE
"import lsdb" fails with a dask error

### ISSUE THREAD

**Issue by hombit** [astronomy-commons/lsdb#285]
**Bug report**

Freshly installed LSDB fails on import:

<details><summary>Expand</summary>

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/hombit/.virtualenvs/lsdb-error/lib/python3.12/site-packages/lsdb/__init__.py", line 2, in <module>
    from .catalog import Catalog
  File "/Users/hombit/.virtualenvs/lsdb-error/lib/python3.12/site-packages/lsdb/catalog/__init__.py", line 1, in <module>
    from .catalog import Catalog
  File "/Users/hombit/.virtualenvs/lsdb-error/lib/python3.12/site-packages/lsdb/catalog/catalog.py", line 6, in <module>
    import dask.dataframe as dd
  File "/Users/hombit/.virtualenvs/lsdb-error/lib/python3.12/site-packages/dask/dataframe/__init__.py", line 100, in <module>
    from dask.dataframe import backends, dispatch, rolling
  File "/Users/hombit/.virtualenvs/lsdb-error/lib/python3.12/site-packages/dask/dataframe/backends.py", line 15, in <module>
    from dask.dataframe.core import DataFrame, Index, Scalar, Series, _Frame
  File "/Users/hombit/.virtualenvs/lsdb-error/lib/python3.12/site-packages/dask/dataframe/core.py", line 36, in <module>
    from dask.dataframe import methods
  File "/Users/hombit/.virtualenvs/lsdb-error/lib/python3.12/site-packages/dask/dataframe/methods.py", line 34, in <module>
    from dask.dataframe.utils import is_dataframe_like, is_index_like, is_series_like
  File "/Users/hombit/.virtualenvs/lsdb-error/lib/python3.12/site-packages/dask/dataframe/utils.py", line 20, in <module>
    from dask.dataframe import (  # noqa: F401 register pandas extension types
  File "/Users/hombit/.virtualenvs/lsdb-error/lib/python3.12/site-packages/dask/dataframe/_dtypes.py", line 9, in <module>
    from dask.dataframe.extensions import make_array_nonempty, make_scalar
  File "/Users/hombit/.virtualenvs/lsdb-error/lib/python3.12/site-packages/dask/dataframe/extensions.py", line 8, in <module>
    from dask.dataframe.accessor import (
  File "/Users/hombit/.virtualenvs/lsdb-error/lib/python3.12/site-packages/dask/dataframe/accessor.py", line 126, in <module>
    class DatetimeAccessor(Accessor):
  File "/Users/hombit/.virtualenvs/lsdb-error/lib/python3.12/site-packages/dask/dataframe/accessor.py", line 81, in __init_subclass__
    _bind_property(cls, pd_cls, attr, min_version)
  File "/Users/hombit/.virtualenvs/lsdb-error/lib/python3.12/site-packages/dask/dataframe/accessor.py", line 35, in _bind_property
    setattr(cls, attr, property(derived_from(pd_cls, version=min_version)(func)))
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/hombit/.virtualenvs/lsdb-error/lib/python3.12/site-packages/dask/utils.py", line 987, in wrapper
    method.__doc__ = _derived_from(
                     ^^^^^^^^^^^^^^
  File "/Users/hombit/.virtualenvs/lsdb-error/lib/python3.12/site-packages/dask/utils.py", line 940, in _derived_from
    method_args = get_named_args(method)
                  ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/hombit/.virtualenvs/lsdb-error/lib/python3.12/site-packages/dask/utils.py", line 701, in get_named_args
    s = inspect.signature(func)
        ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/inspect.py", line 3310, in signature
    return Signature.from_callable(obj, follow_wrapped=follow_wrapped,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/inspect.py", line 3054, in from_callable
    return _signature_from_callable(obj, sigcls=cls,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/inspect.py", line 2642, in _signature_from_callable
    call = _descriptor_get(call, obj)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/inspect.py", line 2467, in _descriptor_get
    return get(descriptor, obj, type(obj))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError
```

</details>

I checked it with CPython 3.11 and 3.12, and with both PyPI release and the current `main` branch.

**Before submitting**
Please check the following:

- [x] I have described the situation in which the bug arose, including what code was executed, information about my environment, and any applicable data others will need to reproduce the problem.
- [x] I have included available evidence of the unexpected behavior (including error messages, screenshots, and/or plots) as well as a descriprion of what I expected instead.
- [ ] If I have a solution in mind, I have provided an explanation and/or pseudocode and/or task list.

**Comment by smcguire-cmu** [comment]
I think what's going on here is a dask dataframe issue on versions of python >3.11.9.  They fix it in [dask version 2024.4.1](https://docs.dask.org/en/stable/changelog.html#v2024-4-1), but this is after the dask expressions update which we are working on with issue #181. So once that is merged and we unpin dask to use later dask versions this issue should be fixed.

**Comment by delucchi-cmu** [comment]
This is the same behavior we saw with this smoke test failure.

https://github.com/astronomy-commons/lsdb/issues/270

With #288, I think we can unpin the python versions.
